src/Model/Item.php

```diff
@@ -1333,88 +1331,80 @@ private static function spool($orig_item)
 		}
 	}
 
-	public static function insert($item, $force_parent = false, $notify = false, $dontcache = false)
+	/**
+	 * Check if the item array is a duplicate
+	 *
+	 * @param array $item
+	 * @return boolean is it a duplicate?
+	 */
+	private static function isDuplicate(array $item)
 	{
-		$orig_item = $item;
-
-		$priority = PRIORITY_HIGH;
-
-		// If it is a posting where users should get notifications, then define it as wall posting
-		if ($notify) {
-			$item['wall'] = 1;
-			$item['origin'] = 1;
-			$item['network'] = Protocol::DFRN;
-			$item['protocol'] = Conversation::PARCEL_DFRN;
-
-			if (is_int($notify)) {
-				$priority = $notify;
-			}
-		} else {
-			$item['network'] = trim(($item['network'] ?? '') ?: Protocol::PHANTOM);
-		}
-
-		$item['guid'] = self::guid($item, $notify);
-		$item['uri'] = substr(Strings::escapeTags(trim(($item['uri'] ?? '') ?: self::newURI($item['uid'], $item['guid']))), 0, 255);
-
-		// Store URI data
-		$item['uri-id'] = ItemURI::insert(['uri' => $item['uri'], 'guid' => $item['guid']]);
-
-		// Store conversation data
-		$item = Conversation::insert($item);
-
-		/*
-		 * If a Diaspora signature structure was passed in, pull it out of the
-		 * item array and set it aside for later storage.
-		 */
-
-		$dsprsig = null;
-		if (isset($item['dsprsig'])) {
-			$encoded_signature = $item['dsprsig'];
-			$dsprsig = json_decode(base64_decode($item['dsprsig']));
-			unset($item['dsprsig']);
+		// Checking if there is already an item with the same guid
+		$condition = ['guid' => $item['guid'], 'network' => $item['network'], 'uid' => $item['uid']];
+		if (self::exists($condition)) {
+			Logger::notice('Found already existing item', [
+				'guid' => $item['guid'],
+				'uid' => $item['uid'],
+				'network' => $item['network']
+			]);
+			return true;
 		}
 
-		$diaspora_signed_text = '';
-		if (isset($item['diaspora_signed_text'])) {
-			$diaspora_signed_text = $item['diaspora_signed_text'];
-			unset($item['diaspora_signed_text']);
+		$condition = ["`uri` = ? AND `network` IN (?, ?) AND `uid` = ?",
```

Comment:
Same question for uri?

Reply:
These whole tests had been created at some time to prevent some situations. I haven't had checked if they still are needed.